orangecontrib/spectroscopy/widgets/owspectra.py

```diff
@@ -1009,6 +1062,101 @@ def grid_apply(self):
         self.plot.showGrid(self.show_grid, self.show_grid, alpha=0.3)
         self.show_grid_a.setChecked(self.show_grid)
 
+    def find_peak_variables(self):
+        if self.maximum_point is None:
+            max_value = []
+            data = np.array(self.data)
+            for i, val in enumerate(data):
+                single_spcetra = data[i]
+                peaks, _ = find_peaks(single_spcetra)
+                max_value.append(np.amax(single_spcetra[peaks]))
+                # method for finding maxes without interference from extra data labels
+            self.maximum_point = np.amax(max_value)
+        if self.minimum_point is None:
+            self.minimum_point = np.amin(self.data)
+        if self.start_point is None:
+            self.start_point = round(np.median(getx(self.data)), 2)
+
+    def peak_apply(self):
+        if self.viewtype == INDIVIDUAL:
+            Label_line = pl.PeakLine()
+            Label_line.setMovable(True)
+            Label_line.setPos(self.start_point)
+            Label_line.label.setText(str(round(self.start_point), 3))
+            Label_line.setPen(pg.mkPen(color=QColor(Qt.black), width=2, style=Qt.DotLine))
+            Label_line.setSpan(mn=self.minimum_point, mx=self.maximum_point)
+            Label_line.label.setColor(color=QColor(Qt.black))
+            Label_line.label.setPosition(1)
+            Label_line.label.setMovable(True)
+            self.plot.addItem(Label_line)
+            Label_line.updateLabel()
+
+    def peak_apply_auto(self, prominence, minHeight, maxHeight, line_overlap):
+        x_axis = getx(self.data)
+        data = np.array(self.data)
```

Comment:
f you need spectral values, there are in data.X. Data can also contain other meta or class data.

Otherwise, self.data_x already contains the sorted x_axis, and data.X[:, self.data_xsind] would give you data with already sorted wavenumbers. I think you find_peaks function expects ordered inputs, while a data table, after transformations in Orange widgets  became unordered wavenumber-wise. Here it is important to use the ordered variants.